interfaces/sourcegen/sourcegen/csharp/_SourceGenerator.py

```diff
@@ -0,0 +1,309 @@
+                    # Special case for the functions that set thermo pairs
+                    # This allows the C# side to pass a pointer to the stack
+                    # Rather than allocating an array on the heap (which requires GC)
+                    param_type = '(double, double)*'
+                else:
+                    param_type = 'double[]'
+
+            params[i] = Param(param_type, param_name)
+
+        func = _CsFunc(ret_type, name, params, del_clazz)
+        self._known_funcs[name] = func
+
+        return func
+
+
+    def _get_property_text(self, clazz: str, c_name: str, cs_name: str):
+        getter = self._known_funcs.get(clazz + "_" + c_name)
+
+        if (getter):
+            # here we have found a simple scalar property
+            prop_type = getter.ret_type
+        else:
+            # here we have found an array-like property (string, double[])
+            getter = self._known_funcs[clazz + "_get" + c_name.capitalize()]
+            # this assumes the last param in the function is a pointer type,
+            # from which we determine the appropriate C# type
+            prop_type = self._prop_type_map[getter.params[-1].p_type]
+
+        setter = self._known_funcs.get(clazz + "_set" + c_name.capitalize())
+
+        if prop_type in ['int', 'double']:
+            text = f'''
+                public {prop_type} {cs_name}
+                {{
+                    get => InteropUtil.CheckReturn(
+                        LibCantera.{getter.name}(_handle));'''
+
+            if(setter):
+                text += f'''
+                    set => InteropUtil.CheckReturn(
+                        LibCantera.{setter.name}(_handle, value));'''
+
+            text += '''
+                }
+            '''
+        elif (prop_type == 'string'):
+            p_type = getter.params[1].p_type
+
+            # for get-string type functions we need to look up the type of the second
+            # (index 1) param for a cast because sometimes it's an int and other times
+            # its a nuint (size_t)
+            text = f'''
+                public unsafe string {cs_name}
+                {{
+                    get => InteropUtil.GetString(40, (length, buffer) =>
+                        LibCantera.{getter.name}(_handle, ({p_type}) length, buffer));
+            '''
+
+            if(setter):
+                text += f'''
+                    set => InteropUtil.CheckReturn(
+                        LibCantera.{setter.name}(_handle, value));'''
+
+            text += '''
+                }
+            '''
+        else:
+            raise ValueError(f'Unable to scaffold properties of type {prop_type}!')
+
+        return(normalize_indent(text))
+
+
+    def generate_source(self, incl_file: os.DirEntry, funcs: list[Func]):
+        cs_funcs = [self._convert_func(f) for f in funcs]
+
+        functions_text = '\n\n'.join((self._get_function_text(f) for f in cs_funcs))
```

Comment:
```suggestion
        functions_text = '\n\n'.join(self._get_function_text(f) for f in cs_funcs)
```
Alternately, I like the style:

```python
... "\n\n".join(map(self._get_function_text, cs_funcs))
```
but that's just a preference.

Reply:
Good to know about the `map()` builtin! I'm also availing myself of `itertools.starmap()`.